kamelet-support/src/main/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteParseCatalog.java

```diff
@@ -0,0 +1,62 @@
+package io.kaoto.backend.metadata.parser.step.camelRoute;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.kaoto.backend.api.metadata.catalog.StepCatalogParser;
+import io.kaoto.backend.metadata.ParseCatalog;
+import io.kaoto.backend.metadata.parser.ClusterParseCatalog;
+import io.kaoto.backend.metadata.parser.GitParseCatalog;
+import io.kaoto.backend.metadata.parser.JarParseCatalog;
+import io.kaoto.backend.metadata.parser.LocalFolderParseCatalog;
+import io.kaoto.backend.model.deployment.kamelet.Kamelet;
+import io.kaoto.backend.model.step.Step;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.nio.file.Path;
+
+@ApplicationScoped
+public final class CamelRouteParseCatalog implements StepCatalogParser {
+
+    @Inject
+    public void setKubernetesClient(final KubernetesClient kubernetesClient) {
+        this.kubernetesClient = kubernetesClient;
+    }
+    private KubernetesClient kubernetesClient;
+
+    private CamelRouteParseCatalog() {
+
+    }
+
+    @Override
+    public ParseCatalog<Step> getParser(final String url, final String tag) {
+        ParseCatalog<Step> parseCatalog = new GitParseCatalog<>(url, tag);
+        parseCatalog.setFileVisitor(new CamelRouteFileProcessor());
+        return parseCatalog;
+    }
+
+
+    @Override
+    public ParseCatalog<Step> getParser(final String url) {
+
+        ParseCatalog<Step> parseCatalog = new JarParseCatalog<>(url);
+        parseCatalog.setFileVisitor(new CamelRouteFileProcessor());
+        return parseCatalog;
+    }
+
+    @Override
+    public ParseCatalog<Step> getParserFromCluster() {
+        ClusterParseCatalog<Step> parseCatalog =
```

Comment:
I think this function should return an empty list directly. We are not storing any `kind` of camel component on the cluster as CRD. So there is nothing you can extract from the cluster to be converted to camel component.

Reply:
It was just the last thing I did. I thought it didn't make any sense